src/tokenizer.rs

```diff
@@ -397,6 +397,17 @@ impl<'a> Tokenizer<'a> {
                 // numbers and period
                 '0'..='9' | '.' => {
                     let mut s = peeking_take_while(chars, |ch| matches!(ch, '0'..='9'));
+
+                    // match binary literal that starts with 0x
+                    if s == "0" && chars.peek() == Some(&'x') {
```

Comment:
Given it appears that postgres seems to treat `0xf00`  as an identifier I wonder if we should check `dialect.is_identifier_start()` prior to matching on this token?

https://dba.stackexchange.com/questions/203358/how-do-i-write-a-hex-literal-in-postgresql

Reply:
In section 4.1.1 (paragraph 2) in the [PostgreSQL docs](https://www.postgresql.org/docs/9.1/sql-syntax-lexical.html) they say that the identifiers have to start with a letter or an underscore.

Reply:
🤔  I tried this with postgres and the example `select 0xCC` still parses strangely, but as you mention, using `0x` as column names without quoting them doesn't work

```shell
(arrow_dev) alamb@MacBook-Pro:~/Software/arrow-datafusion$ psql
psql (13.4)
Type "help" for help.

alamb=# SELECT 0xCC;
 xcc 
-----
   0
(1 row)

alamb=# create table foo(0xCC VARCHAR);
ERROR:  syntax error at or near "0"
LINE 1: create table foo(0xCC VARCHAR);

alamb=# select c1 as 0x44 from test ;
ERROR:  syntax error at or near "0"
LINE 1: select c1 as 0x44 from test ;

alamb=# select c1 as "0x44" from test ;
 0x44 
------
    0
    3
     
    1
    3
(5 rows)


```